### PR TITLE
MAINT: pytest syntax, NEP29, github actions

### PR DESCRIPTION
**Pull request by jklenzing**
# Description

Addresses #1053, #1048

Updates to remove nose syntax.  pytest will remove support for this in the future.

Also updates github actions and NEP29 implementation.

## Type of change

- Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?

pytest

**Test Configuration**:
* Operating system: Mac OS Monterrey
* Version number: Python 3.8.11

# Checklist:

- [x] Make sure you are merging into the ``develop`` (not ``main``) branch
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] Add a note to ``CHANGELOG.md``, summarizing the changes

If this is a release PR, replace the first item of the above checklist with the release
checklist on the wiki: https://github.com/pysat/pysat/wiki/Checklist-for-Release

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -21,16 +21,18 @@ jobs:
...
-      uses: actions/setup-python@v2
+      uses: actions/setup-python@v4
       with:
         python-version: ${{ matrix.python-version }}
 
     - name: Install NEP29 dependencies
       if: ${{ matrix.numpy_ver != 'latest'}}
       run: |
         pip install --no-binary :numpy: numpy==${{ matrix.numpy_ver }}
+        # Need to force install of pandas compliant with NEP29
```
Comment (rstoneback) [inline .github/workflows/main.yml:34]: I thought we just added support for pandas 1.5....

**Comment by rstoneback** [comment]
Bummer to have to change all the setup calls to setup_method....  don't really see how it is better. Anyway, guess I'll have to do this for OMMBV and others coming up then too. Thanks for taking care of this here @jklenzing.

File: .github/workflows/main.yml (hunk 1 above)
Comment (jklenzing) [reply .github/workflows/main.yml:34]: We support pandas 1.5, but pandas 1.5 does not support numpy 1.20.  It's a problem created by pandas not supporting the NEP29 schedule that goes away Jan 31.

**Comment by jklenzing** [comment]
> Bummer to have to change all the setup calls to setup_method.... don't really see how it is better. Anyway, guess I'll have to do this for OMMBV and others coming up then too. Thanks for taking care of this here @jklenzing.

I think the rationale is that it's more clear.  pytest distinguishes between `setup_method` and `setup_class`, which get called at different points during the tests.  The calls to `setup` were built for backwards compatibility to nose, which they don't plan to support in the future.

**Review by rstoneback — APPROVED** [review]
Thanks for the clarification and keeping pysat up to date.

**Pull request merged**